healthcheck/handler.go

```diff
@@ -0,0 +1,109 @@
+package healthcheck
+
+import (
+	"encoding/json"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/pkg/errors"
+	"net/http"
+	"time"
+)
+
+const (
+	StatusOK       = "OK"
+	StatusCritical = "CRITICAL"
+	StatusWarning  = "WARNING"
+)
+
+// Handler returns the current health check of the app
+func (hc HealthCheck) Handler(w http.ResponseWriter, req *http.Request) {
+	now := time.Now().UTC()
+	ctx := req.Context()
+
+	var checks []Check
+
+	for _, client := range hc.Clients {
+		if client.Check != nil {
```

Comment:
I think we should log an error with the name of the check if the client is nil - make it obvious that its not all running as expected.

Reply:
As per the initial design concept, we don't always know the name of the app as this is only within the `check`,  if this is the first check that has failed then `check` is `nil`. A workaround would be to assign the `client` the name of the app check - this could be accomplished by the user providing it when they call `NewClient`.